cpp/order/Steinhardt.cc

```diff
@@ -30,151 +28,154 @@ void Steinhardt::computeYlm(const float theta,
+    }
+    if (m_Wl)
+    {
+        memset((void*) m_WliOrder.get(), 0, sizeof(complex<float>) * m_Np);
+    }
+}
+
+void Steinhardt::compute(const box::Box& box, const locality::NeighborList* nlist, const vec3<float>* points,
+                         unsigned int Np)
+{
+    // Allocate and zero out arrays as necessary
+    reallocateArrays(Np);
 
     // Computes the base Qlmi required for each specialized order parameter
-    Steinhardt::baseCompute(box, nlist, points);
+    baseCompute(box, nlist, points);
 
     if (m_average)
-        {
-        Steinhardt::computeAve(box, nlist, points);
-        }
+    {
+        computeAve(box, nlist, points);
+    }
 
-	// Reduce Qlm
-	Steinhardt::reduce();
+    // Reduce Qlm
+    reduce();
 
     if (m_Wl)
-        {
-        memset((void*) m_WliOrder.get(), 0, sizeof(complex<float>)*m_Np);
+    {
         if (m_average)
-            {
-            Steinhardt::aggregateWl(m_WliOrder, m_QlmiAve);
-			m_NormWl = Steinhardt::normalizeWl();
-            }
-        else
-            {
-            Steinhardt::aggregateWl(m_WliOrder, m_Qlmi);
-			m_NormWl = Steinhardt::normalizeWl();
-            }
+        {
+            aggregateWl(m_WliOrder, m_QlmiAve);
+            m_NormWl = normalizeWl();
         }
+        else
         {
+            aggregateWl(m_WliOrder, m_Qlmi);
+            m_NormWl = normalizeWl();
+        }
+    }
+    {
         if (m_average)
-            {
-            m_Norm = Steinhardt::normalize();
-            }
+        {
+            m_Norm = normalize();
+        }
         else
-            {
-            m_Norm = Steinhardt::normalize();
-            }
+        {
+            m_Norm = normalize();
         }
     }
+}
 
-void Steinhardt::baseCompute(const box::Box& box,
-							 const locality::NeighborList *nlist,
-							 const vec3<float> *points)
-    {
+void Steinhardt::baseCompute(const box::Box& box, const locality::NeighborList* nlist,
+                             const vec3<float>* points)
+{
     nlist->validate(m_Np, m_Np);
 
-    memset((void*) m_Qlmi.get(), 0, sizeof(complex<float>)*(2*m_l+1)*m_Np);
-    memset((void*) m_Qli.get(), 0, sizeof(float)*m_Np);
-
-    parallel_for(tbb::blocked_range<size_t>(0, m_Np),
-        [=] (const blocked_range<size_t>& r)
-        {
+    parallel_for(tbb::blocked_range<size_t>(0, m_Np), [=](const blocked_range<size_t>& r) {
         const float rminsq = m_rmin * m_rmin;
         const float rmaxsq = m_rmax * m_rmax;
-        const float normalizationfactor = 4*M_PI/(2*m_l+1);
-        const size_t *neighbor_list(nlist->getNeighbors());
-
-		if (!m_average)
-			{
-			bool Qlm_exists;
-			m_Qlm_local.local(Qlm_exists);
-			if (!Qlm_exists)
-				{
-				m_Qlm_local.local() = new complex<float> [2*m_l+1];
-				memset((void*) m_Qlm_local.local(), 0, sizeof(complex<float>) * (2*m_l+1));
-				}
-			}
+        const float normalizationfactor = 4 * M_PI / (2 * m_l + 1);
+        const size_t* neighbor_list(nlist->getNeighbors());
+
```

Comment:
Add comment: `// Initialize thread-local m_Qlm and compute it in this function, if we won't average over neighbors later`

Reply:
Add a similar comment below in the average method, something like `// Initialize thread-local m_Qlm for storing neighbor-averaged QlmiAve, reduced over particles`